packages/react-data-grid/src/utils/columnUtils.ts

```diff
@@ -64,22 +65,77 @@ export function getColumnMetrics<R>(metrics: Metrics<R>): ColumnMetrics<R> {
   };
 }
 
+function getMinWidthFromViewportWidth(
+  minWidth: string,
+  viewportWidth: number
+): number {
+  return Math.floor((viewportWidth * parseInt(minWidth, 10)) / 100);
+}
+
+function getSpecifiedMinColumnWidth(
+  viewportWidth: number,
+  minColumnWidth: number,
+  width?: number | string
+): number {
+  switch (typeof width) {
+    case 'string':
+      return /^\d+%$/.test(width)
+        ? getMinWidthFromViewportWidth(width, viewportWidth)
+        : minColumnWidth;
+    case 'number':
+      return width;
+    default:
+      return minColumnWidth;
+  }
+}
+
+function getMinColumnWidthSpecified<R>(
+  column: Column<R>,
+  viewportWidth: number,
+  minColumnWidth: number
+): number {
+  // don't evaluate if minWidth is lower than width for SelectColumn
+  if (column.key === SelectColumn.key) {
+    return getSpecifiedMinColumnWidth(
+      viewportWidth,
+      minColumnWidth,
+      column.width // based on column width
+    );
+  }
+  const minColumnWidthSpecified = getSpecifiedMinColumnWidth(
+    viewportWidth,
+    minColumnWidth,
+    column.minWidth // based on column minWidth, instead of using grid minColumnWidth
+  );
+  return Math.min(minColumnWidthSpecified, minColumnWidth);
+}
+
 function getSpecifiedWidth<R>(
   column: Column<R>,
   columnWidths: Map<keyof R, number>,
   viewportWidth: number,
   minColumnWidth: number
 ): number | void {
+  // get minColumnWidth from column prop or grid prop
+  const minColumnWidthSpecified: number = getMinColumnWidthSpecified(
+    column,
```

Comment:
We will only use `column.minWidth` for getting the `minWidth`, so it's better to be specific like 
```suggestion
  const minWidth = getMinWidth(column.minWidth, defaultMinColumnWidth);
```
I also reduced the noise words in the `function` and `variable` to be more concise.

In addition, in the `getMinWidth`, I believe we need to refine the definition for which field to use, as I believe `column.minWidth` and `grid.minColumnWidth` should be precedence relationship rather than comparison relationship, which means when `column.minWidth` is available while it's greater than `0`, we should just return it for the `minWidth` we will be using for the rest. Hence, `getMinWidth` will be like the following
```javascript
function getMinWidth(columnMinWidth, defaultMinColumnWidth): number {
  if (columnMinWidth != null && columnMinWidth > 0) {
    return columnMinWidth;
  }

  return defaultMinColumnWidth;
}
```
And we can further refine it to be just an one line statement with tertiary operator.
```javascript
function getSpecifiedWidth<R>(
  column: Column<R>,
  columnWidths: Map<keyof R, number>,
  viewportWidth: number,
	defaultMinColumnWidth: number
): number | void {
const minWidth = column.minWidth  != null && column.minWidth > 0 ? column.minWidth : defaultMinColumnWidth;

...
```

Reply:
sounds good

Reply:
- on load `width` value will be the column width if above `minWidth` value (if `minWidth` undefined, ... above `minColumnWidth` value) otherwise it will be `minWidth || minColumnWidth`
- on resize the width will go down until one of the `minWidth` or `minColumnWidth` is reached
- checkbox SelectColumn can have its `width` re-defined, if by any reason the `width` is undefined, it will fallback on `minWidth`